clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11774,6 +11774,9 @@ def err_omp_clause_requires_dispatch_construct : Error<
   "'%0' clause requires 'dispatch' context selector">;
 def err_omp_append_args_with_varargs : Error<
   "'append_args' is not allowed with varargs functions">;
+def warn_omp_dispatch_clause_novariants_nocontext : Warning<
+  "only 'novariants' clause is supported when 'novariants' & 'nocontext' clauses occur on the same dispatch construct">,
```

Comment:
1.
```suggestion
  "only 'novariants' clause is expected">,
```
or something like this.

2. No test for this warning

Reply:
In case of 
       `#pragma omp dispatch nocontext(c1) novariants(c2) depend(inout:x)`
the alternative warning message that you have suggested may not be proper because it indicates `depend` is also not expected.

Reply:
I have added extra tests in clang/test/OpenMP/dispatch_messages.cpp.

Reply:
We do not use such format of the warnings, please check the wording for other warnings and try to replicate what standard actual says

Reply:
I have re-worded the warning message to:
`'nocontext' clause is ignored, only 'novariants' clause is applied`
 Please check.